pkg/cmd/cli/snapshotlocation/set.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright The Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package snapshotlocation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	corev1api "k8s.io/api/core/v1"
+
+	kbclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/client"
+	"github.com/vmware-tanzu/velero/pkg/cmd"
+	"github.com/vmware-tanzu/velero/pkg/cmd/util/flag"
+	"github.com/vmware-tanzu/velero/pkg/cmd/util/output"
+)
+
+func NewSetCommand(f client.Factory, use string) *cobra.Command {
+	o := NewSetOptions()
+
+	c := &cobra.Command{
+		Use:   use + " NAME",
+		Short: "Set specific features for a snapshot location",
+		Args:  cobra.ExactArgs(1),
+		Run: func(c *cobra.Command, args []string) {
+			cmd.CheckError(o.Complete(args, f))
+			cmd.CheckError(o.Validate(c, args, f))
+			cmd.CheckError(o.Run(c, f))
+		},
+	}
+
+	o.BindFlags(c.Flags())
+	return c
+}
+
+type SetOptions struct {
+	Name       string
+	Credential flag.Map
+}
+
+func NewSetOptions() *SetOptions {
+	return &SetOptions{}
```

Comment:
We need to initialize the `Credential` field as is done in [3190](https://github.com/vmware-tanzu/velero/pull/3190/files#diff-da0544a77f284ebf456c090aeae8e1b5d0100cbabc631d2c96dc18365dfd8f70R62), otherwise there is an error when attempting to set the credential field: 
```
An error occurred: invalid argument "foo=bar" for "--credential" flag: error parsing "f"
```

Reply:
Fixed!